lib/shopify_app/session/session_repository.rb

```diff
@@ -2,21 +2,23 @@ module ShopifyApp
   class SessionRepository
     class ConfigurationError < StandardError; end
 
+    EXPECTED_METHODS = [:store, :retrieve].freeze
+    EXPECTED_SHOP_STORAGE_METHODS = (EXPECTED_METHODS + [:retrieve_by_shopify_domain]).freeze
+    EXPECTED_USER_STORAGE_METHODS = (EXPECTED_METHODS + [:retrieve_by_shopify_user_id]).freeze
+
     class << self
       def shop_storage=(storage)
         @shop_storage = storage
+        return unless storage
 
-        unless storage.nil? || self.shop_storage.respond_to?(:store) && self.shop_storage.respond_to?(:retrieve)
-          raise ArgumentError, "shop storage must respond to :store and :retrieve"
-        end
+        raise ArgumentError, "shop storage does not have expected interface" unless expected_shop_interface?(shop_storage)
```

Comment:
Re: Tim's comment about changing the interface; it does occur to me that this changes the interface (we now require an extra method). Maybe `expected_shop_interface?` (and the user one) should only expect the new `retrieve_` methods if the `allow_jwt_authentication` option is active.

Reply:
@ragalie Actually I am about to ship https://github.com/Shopify/shopify_app/pull/925 which will remove that

The problem is that it loads the session constants at startup and this breaks zeitworks reloading functionality, thus raises a deprecation warning

Reply:
:+1:

Reply:
@rezaansyed we should wait to ship until #925 ships in this case and then rebase ours onto theirs.

Reply:
Sounds good. I'll wait for that PR to ship. 👍